src/screens/User/User.tsx

```diff
@@ -40,6 +39,9 @@ const User = (): JSX.Element => {
 
   const onCardClick = (playlistItem: PlaylistItem) => history.push(cardUrl(playlistItem));
   const onCardHover = (playlistItem: PlaylistItem) => updateBlurImage(playlistItem.image);
+  const onLogout = useCallback(() => {
+    logout();
```

Comment:
Did the history.push get dropped here?  Is it possible to just move that to the account store logout function (probably using `window.location.href = '/'` to avoid problems with hooks)

Reply:
Ah, I see that and it certainly should work as you describe.  Do you think it might be easier to read / follow / update the code if the navigation lived inside the logout function though?  For example if someone wants to change the post-logout page or let users stay on the same page after logout?  Alternatively, if you want to leave the navigation in the hook, can you add a comment that this returns the user to the main page after logout and also remove the history.push in UserMenu?

Reply:
@dbudzins thank you for ideas! I modified the solution a bit and created a new Provider to control page access. And it also does redirect to the base page if the user is not defined. What do you think about this approach?